gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/heat_node_driver.py

```diff
@@ -387,6 +387,15 @@ def _update_template_with_firewall_rules(self, context, provider_ptg,
                             stack_template[resources_key],
                             is_template_aws_version,
                             resource_name)
+        fw_resource_name = 'OS::Neutron::Firewall'
+        fw_key = self._get_heat_resource_key(
+                            stack_template[resources_key],
+                            is_template_aws_version,
+                            fw_resource_name)
+        # Include provider name in firewall, firewall policy.
+        ptg_name = '-' + provider_ptg['name']
+        stack_template[resources_key][fw_policy_key][properties_key]['name'] += ptg_name
+        stack_template[resources_key][fw_key][properties_key]['name'] += ptg_name
 
```

Comment:
please make this modular by introducing a method to update name  like the lb case.
Then the same method can be re-used by OC service node driver